src/main/java/io/zksync/ethereum/EthereumProvider.java

```diff
@@ -0,0 +1,87 @@
+package io.zksync.ethereum;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.DefaultBlockParameterName;
+import org.web3j.protocol.core.methods.response.EthGetBalance;
+import org.web3j.protocol.core.methods.response.EthGetTransactionCount;
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import org.web3j.tx.Transfer;
+import org.web3j.tx.gas.ContractGasProvider;
+import org.web3j.tx.gas.StaticGasProvider;
+import org.web3j.utils.Convert.Unit;
+
+import io.zksync.domain.token.Token;
+import io.zksync.ethereum.wrappers.ERC20;
+import io.zksync.ethereum.wrappers.ZkSync;
+import io.zksync.signer.EthSigner;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class EthereumProvider {
+
+    private static final BigInteger DEFAULT_THRESHOLD = BigInteger.valueOf(2).pow(255);
+    private static final ContractGasProvider DEFAULT_GAS_PROVIDER = new StaticGasProvider(BigInteger.ZERO,
+            BigInteger.ZERO);
+
+    private final Web3j web3j;
+    private final EthSigner ethSigner;
+    private final ZkSync contract;
+
+    public CompletableFuture<TransactionReceipt> approveDeposits(Token token, Optional<BigInteger> limit) {
+        ERC20 tokenContract = ERC20.load(token.getAddress(), this.web3j, this.ethSigner.getCredentials(), this.contract.getGasProvider());
+        return tokenContract.approve(this.contractAddress(), limit.orElse(DEFAULT_THRESHOLD)).sendAsync();
```

Comment:
The default limit should not be equal to `DEFAULT_THRESHOLD`. There is another constant for that https://github.com/matter-labs/zksync/blob/5b936b1855a08033cca7f75d6f87fde106c6e8fd/sdk/zksync.js/src/utils.ts#L17

It is very important because each time the user deposits some tokens, the allowance is reduced. Thus, if we approve only `DEFAULT_THRESHOLD` tokens, then right after the first deposit, the function `isDepositApproved` will always return false.